src/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -39,10 +39,10 @@ public static IQueryable<TSource> Where<TSource>(this IQueryable<TSource> source
             return source.Provider.CreateQuery<TSource>(
                 Expression.Call(
                     null,
-                    GetMethodInfoOf(() => Queryable.Where(
+                    GetMethodInfoOf(() => Where(
```

Comment:
As I remember, the variant based on ```.Method``` had some issues with .Net native (or CoreCLR reduced API at the time.

Some form of caching might be useful here indeed.